### PR TITLE
Updates all the way to 19.0.8

### PR DESCRIPTION
**Pull request by theolivenbaum**
Hi @GregorBiswanger , just saw that you merged my other PR.

If you want, here are the remaining changes I've done since then - lots of fixes, updates, etc...

### CODE REVIEW

**Comment by theolivenbaum** [comment]
I'll see if I can clean up the merge for this PR first, so no hurry. Just wanted to open it so we can check what changed

File: ElectronNET.API/App.cs [hunk 1]
```diff
@@ -495,9 +586,13 @@ public string Name
         public Task<string> GetNameAsync() => BridgeConnector.OnResult<string>("appGetName", "appGetNameCompleted");
 
 
-        internal App() 
+        private App()
         {
-            CommandLine = new CommandLine();
+            if (OperatingSystem.IsMacOS() || OperatingSystem.IsLinux())
+            {
+                AppContext.SetSwitch("System.Drawing.EnableUnixSupport", true);
```
Comment (danatcofo) [inline ElectronNET.API/App.cs:593]: this is no longer required with the move away from `System.Drawing.Common` done in PR #658

File: ElectronNET.API/App.cs [hunk 2]
```diff
@@ -1244,7 +1396,14 @@ internal void PreventQuit()
         /// <param name="fn">The handler</param>
         public void Once(string eventName, Action<object> fn) => Events.Instance.Once(ModuleName, eventName, fn);
 
-        private readonly JsonSerializer _jsonSerializer = new JsonSerializer()
+
+        /// <summary>
+        /// If you're using a splashscreen in the electron.manifest.json, the window will ony be fully destroyed once you call this method once.
+        /// You should only do this after creating another window, to avoid a bug where the Electron renderer process frezees till any window interaction.
+        /// </summary>
+        public void DestroySplashScreen() =>  BridgeConnector.Emit("splashscreen-destroy");
```
Comment (danatcofo) [inline ElectronNET.API/App.cs:1404]: This should definitely be called out in any documentation as a new unexpected requirement or something.

File: ElectronNET.API/BridgeConnector.cs [hunk 3]
```diff
@@ -195,7 +332,7 @@ public static async Task<T> OnResult<T>(string triggerEvent, string completedEve
                         EventTasks<T>.DoneWith(completedEvent, eventKey, taskCompletionSource);
                     });
 
-                    Emit(triggerEvent, args);
+                    await EmitAsync(triggerEvent, args);
```
Comment (danatcofo) [inline ElectronNET.API/BridgeConnector.cs:335]: `await EmitAsync(triggerEvent, args).ConfigureAwait(false);`

every async should call `ConfigureAwait(false);` where possible in the method chains.

File: ElectronNET.API/BrowserWindow.cs [hunk 4]
```diff
@@ -1468,21 +1535,22 @@ public void SetPosition(int x, int y)
...
+                x -= 7;
             }
 
             BridgeConnector.Emit("browserWindowSetPosition", Id, x, y, animate);
         }
 
         private bool isWindows10()
         {
-            return RuntimeInformation.OSDescription.Contains("Windows 10");
+            return OperatingSystem.IsWindowsVersionAtLeast(10);
```
Comment (danatcofo) [inline ElectronNET.API/BrowserWindow.cs:1553]: 👍

File: ElectronNET.API/Clipboard.cs [hunk 5]
```diff
@@ -170,5 +177,12 @@ public void WriteImage(NativeImage image, string type = "")
         {
             BridgeConnector.Emit("clipboard-writeImage", JsonConvert.SerializeObject(image), type);
         }
+
+        private static readonly JsonSerializer _jsonSerializer = new()
```
Comment (danatcofo) [inline ElectronNET.API/Clipboard.cs:181]: This pattern of serializer is used so pervasively in the project that it would be beneficial just to have have a global static one defined and referenced.

File: ElectronNET.API/Entities/BrowserWindowOptions.cs [hunk 6]
```diff
@@ -36,62 +37,62 @@ public class BrowserWindowOptions
...
         [DefaultValue(true)]
-        public bool Movable { get; set; } = true;
+        public bool? Movable { get; set; } = true;
 
         /// <summary>
         /// Whether window is minimizable. This is not implemented on Linux. Default is true.
         /// </summary>
         [DefaultValue(true)]
-        public bool Minimizable { get; set; } = true;
+        public bool? Minimizable { get; set; } = true;
```
Comment (danatcofo) [inline ElectronNET.API/Entities/BrowserWindowOptions.cs:83]: Where we have default values... does it make sense to have the type be nullable?

File: ElectronNET.API/Entities/NativeImage.cs [hunk 7]
```diff
@@ -16,10 +16,12 @@ namespace ElectronNET.API.Entities
     [JsonConverter(typeof(NativeImageJsonConverter))]
```
Comment (danatcofo) [inline ElectronNET.API/Entities/NativeImage.cs:16]: There will be a lot of conflicts here with #658

File: ElectronNET.API/IpcMain.cs [hunk 8]
```diff
@@ -62,6 +64,38 @@ public void On(string channel, Action<object> listener)
             });
         }
 
+        /// <summary>
+        ///  Listens to channel, when a new message arrives listener would be called with 
+        ///  listener(event, args...). This listner will keep the window event sender id
+        /// </summary>
+        /// <param name="channel">Channelname.</param>
+        /// <param name="listener">Callback Method.</param>
+        public void OnWithId(string channel, Action<(int browserId, int webContentId, object arguments)> listener)
```
Comment (danatcofo) [inline ElectronNET.API/IpcMain.cs:73]: I really like this usecase.  In my prefix implementation I generisize my listeners with the expectation that the windowId/browserId is sent as part of the packet to accomplish the same goal of directing the message to the correct object instance. 

This usecase is really good for the single page instance.. i.e. there is only every one instance of a page.  I think it starts to break down when you have multiple instances of the same page however because you have to register the channel over and over again rather than having a single generalized version.

But I think this covers a really important usecase.  well thought out!

File: ElectronNET.WebApp/ElectronNET.WebApp.csproj [hunk 9]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
     <RuntimeIdentifiers>win-x64</RuntimeIdentifiers>
```
Comment (danatcofo) [inline ElectronNET.WebApp/ElectronNET.WebApp.csproj:4]: not part of this PR but do we really need the RuntimeIdentifiers here?  I've run the app in macos just fine as well..

File: ElectronNET.WebApp/ElectronNET.WebApp.csproj [hunk 10]
```diff
@@ -16,6 +16,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="5.0.0" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
+    <PackageReference Include="runtime.osx.10.10-x64.CoreCompat.System.Drawing" Version="6.0.5.128" />
```
Comment (danatcofo) [inline ElectronNET.WebApp/ElectronNET.WebApp.csproj]: invalidated by #658 I believe

**Comment by danatcofo** [comment]
I really like this PR.  As you said it needs to be resolved with all the other stuff that @GregorBiswanger merged the other day and updated accordingly so I tried to add comments and reviews with that in mind.  

Some things I'd like to call attention to:

* It doesn't appear to extend the support of DI to some of the new classes you put together or methods that have been added.  As someone who uses DI extensively and doesn't really like the static pattern promoted here as a downstream access method I would really like to see some refactoring to accommodate DI patterns as has been implemented in other PR's.
* Places where you implemented an async method are missing standard hardening code such as `ConfigureAwait(false);`. I didn't peruse the entire codebase before to see if that was endemic but I noticed it in your changes.
* this code is riddled with `[SupportedOSPlatform("macos")]` for example.  I like this a lot but wonder if we should refactor the string parameter to a constant such as 'OsPlatforms.MacOs', 'OsPlatforms.Windows', etc... rather than peppering a duplicate string everywhere. 
* It appears your ts/js styling and linting rules may be different than others given the number of places where the difference is newlines and spaces only.  Should we add the styling and linting rules as part of the project to minimize churn?

On a completely different note.... 

This project uses TS as its source javascirpt files, yet we still put the maps and output js files into source control.  I really feel we shouldn't do that give the churn that they will go through that is entirely redundant to the the changes in source files.  Maybe given the scope of this PR it would be a good idea to remove those files and add them to the .gitignore for that folder.

Looking forward to your feedback and discussion

**Comment by GregorBiswanger** [comment]
I would like to merge your pull requests.. can you please check the conflicted ones.

**Comment by GregorBiswanger** [comment]
@theolivenbaum I would like to merge your pull requests.. can you please check the conflicted ones.

**Comment by theolivenbaum** [comment]
Hi @GregorBiswanger - I unfortunately don't have the bandwidth to review this now - specially to make sure this doesn't introduce any new issues due to the big changes I added on our fork to be async-only.

**Comment by lafleurh** [comment]
@GregorBiswanger  Let me know what help you need bringing this in. We are looking to update our branch and contribute some WiX stuff back. I am not able to build or run the latest code.

**Comment by GregorBiswanger** [comment]
@lafleurh There are some merge conflicts. I only have a few weeks time.

**Comment by GregorBiswanger** [comment]
@theolivenbaum I merged all and the project should build without problems. I don´t know why you implement the SocketIO source directly?! I was think, you wont to use the nuge library?! Please check the current merged version on our repo.. Thanks!

**Comment by GregorBiswanger** [comment]
@theolivenbaum I get the following exception when running a new "Hello World" project.

I had installed the Nito.AsyncEx.Coordination Library via NuGet at the API project. Why do we need this dependency? Can you do without?

```
electron.cmd "..\..\main.js"

Electron Socket IO Port: 8000
Electron Socket started on port 8000 at 127.0.0.1
.NET Core Port: 8001
stderr: Unhandled exception.
stderr: Unhandled exception.
stderr: System.TypeInitializationException: The type initializer for 'ElectronNET.API.BridgeConnector' threw an exception.
 ---> System.IO.FileNotFoundException: Could not load file or assembly 'Nito.AsyncEx.Coordination, Version=5.1.2.0, Culture=neutral, PublicKeyToken=null'. Das System kann die angegebene Datei nicht finden.
File name: 'Nito.AsyncEx.Coordination, Version=5.1.2.0, Culture=neutral, PublicKeyToken=null'
   at ElectronNET.API.BridgeConnector..cctor()
   --- End of inner exception stack trace ---
   at ElectronNET.API.BridgeConnector.get_Logger() in C:\Users\Gregor\source\repos\Electron.NET\ElectronNET.API\BridgeConnector.cs:line 536
   at ElectronNET.API.BridgeConnector.Log(String formatString, Object[] args) in C:\Users\Gregor\source\repos\Electron.NET\ElectronNET.API\BridgeConnector.cs:line 396
   at ElectronNET.API.WebHostBuilderExtensions.UseElectron(IWebHostBuilder builder, String[] args) in C:\Users\Gregor\source\repos\Electron.NET\ElectronNET.API\WebHostBuilderExtensions.cs:line 26
   at Program.<Main>$(String[] args) in C:\Users\Gregor\source\repos\ElectronNetNewTest\Program.cs:line 5
   at Program.<Main>(String[] args)

stderr: System.TypeInitializationException: The type initializer for 'ElectronNET.API.BridgeConnector' threw an exception.
 ---> System.IO.FileNotFoundException: Could not load file or assembly 'Nito.AsyncEx.Coordination, Version=5.1.2.0, Culture=neutral, PublicKeyToken=null'. Das System kann die angegebene Datei nicht finden.
File name: 'Nito.AsyncEx.Coordination, Version=5.1.2.0, Culture=neutral, PublicKeyToken=null'
   at ElectronNET.API.BridgeConnector..cctor()
   --- End of inner exception stack trace ---
   at ElectronNET.API.BridgeConnector.get_Logger() in C:\Users\Gregor\source\repos\Electron.NET\ElectronNET.API\BridgeConnector.cs:line 536
   at ElectronNET.API.BridgeConnector.Log(String formatString, Object[] args) in C:\Users\Gregor\source\repos\Electron.NET\ElectronNET.API\BridgeConnector.cs:line 396
   at ElectronNET.API.WebHostBuilderExtensions.UseElectron(IWebHostBuilder builder, String[] args) in C:\Users\Gregor\source\repos\Electron.NET\ElectronNET.API\WebHostBuilderExtensions.cs:line 26
   at Program.<Main>$(String[] args) in C:\Users\Gregor\source\repos\ElectronNetNewTest\Program.cs:line 5
   at Program.<Main>(String[] args)

.NET process exited with code 3762504530
Will quit Electron, as exit code != 0 (got 3762504530)
```

**Comment by GregorBiswanger** [comment]
@theolivenbaum Can you please see the error message. ☝️

**Comment by lafleurh** [comment]
I managed to get it working for us. See https://github.com/evoleap/Electron.NET/tree/master19 bit I'm waiting on another project and we have some breaking changes to rectify. Also, we went to .net 6.
